opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -198,21 +198,23 @@ def shutdown(self):
             _logger.warning("shutdown can only be called once")
             return False
 
-        result = True
+        overall_result = True
 
         for metric_reader in self._sdk_config.metric_readers:
-            result = result and metric_reader.shutdown()
+            metric_reader_result = metric_reader.shutdown()
 
-            if not result:
+            if not metric_reader_result:
                 _logger.warning("A MetricReader failed to shutdown")
```

Comment:
can we log the specific one that failed?

Reply:
Done, not sure if this will be of much help since there is nothing readable that identifies a metric reader, but maybe that can be added in the future as an implementation of `__repr__`.